projects/output-and-error-handling/server/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+func main() {
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		// We generate a random number between 0 and 9 (inclusive), so that we can decide whether to behave properly (half of the time), or simulate error conditions.
+		randomNumber := rand.Intn(10)
+		if randomNumber < 5 {
+			// 50% of the time, we just report the weather.
+			if randomNumber < 3 {
+				w.Write([]byte("Today it will be sunny!\n"))
+			} else {
+				w.Write([]byte("I'd bring an umbrella, just in case...\n"))
+			}
+		} else if randomNumber < 8 {
```

Comment:
this is a very minor nitpick and i fully admit this is a style preference thing, but I don't love the implicitness of the >=5 from the initial if clause here. just makes the logic a bit harder to follow IMO
would this be a little cleaner as a switch rather than an if?

Reply:
Absolutely, and completely agree! Done.

(Two of the really big things I wish Go had were proper enum types, and switch/match expressions rather than statements :()